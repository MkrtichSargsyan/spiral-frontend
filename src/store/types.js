export const FETCH_AGENTS_START = 'FETCH_AGENTS_START';
export const FETCH_AGENTS_SUCCESS = 'FETCH_AGENTS_SUCCESS';
export const FETCH_AGENTS_ERROR = 'FETCH_AGENTS_ERROR';
export const CHOOSE_AGENT = 'CHOOSE_AGENT';
export const FETCH_AGENT_HOUSES = 'FETCH_AGENT_HOUSES';
export const FETCH_AGENTS_HOUSES_SUCCESS = 'FETCH_AGENTS_HOUSES_SUCCESS';
// houses types
export const FETCH_HOUSES_START = 'FETCH_HOUSES_START';
export const FETCH_HOUSES_SUCCESS = 'FETCH_HOUSES_SUCCESS';
export const FETCH_HOUSES_ERROR = 'FETCH_HOUSES_ERROR';
export const CHOOSE_HOUSE = 'CHOOSE_HOUSE';
export const FETCH_HOUSE_BY_ID = 'FETCH_HOUSE_BY_ID';
// export const OPEN_MODAL = 'OPEN_MODAL';
// export const CLOSE_MODAL = 'CLOSE_MODAL';
// export const SAVE_USER = 'SAVE_USER';
// export const SAVE_TOKEN = 'SAVE_TOKEN'
