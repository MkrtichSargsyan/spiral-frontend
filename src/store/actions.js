import * as types from '../store/types';

import axios from 'axios';

export const fetchAgents = (url) => {
  return async (dispatch) => {
    dispatch(fetchAgentsStart());

    try {
      const response = await axios.get(url);
      dispatch(fetchAgentsSuccess(response.data));
    } catch (e) {
      dispatch(fetchAgentsError(e));
    }
  };
};

const fetchAgentsStart = () => {
  return {
    type: types.FETCH_AGENTS_START,
  };
};

const fetchAgentsSuccess = (agentsList) => {
  return {
    type: types.FETCH_AGENTS_SUCCESS,
    agentsList,
  };
};

const fetchAgentsError = (e) => {
  return {
    type: types.FETCH_AGENTS_ERROR,
    error: e,
  };
};

// choose agent

export const chooseAgent = (choosedAgent) => {
  return {
    type: types.CHOOSE_AGENT,
    choosedAgent,
  };
};

// fetch agent houses

export const fetchAgentHouses = (url) => {
  return async (dispatch) => {
    try {
      const response = await axios.get(url);
      dispatch(fetchAgentHousesSuccess(response.data));
    } catch (e) {
      console.log('show error');
    }
  };
};

const fetchAgentHousesSuccess = (agentHousesList) => {
  return {
    type: types.FETCH_AGENTS_HOUSES_SUCCESS,
    agentHousesList,
  };
};

// fetch houses

export const fetchHouses = (url) => {
  return async (dispatch) => {
    dispatch(fetchHousesStart());

    try {
      const response = await axios.get(url);
      dispatch(fetchHousesSuccess(response.data));
    } catch (e) {
      dispatch(fetchHousesError(e));
    }
  };
};

const fetchHousesStart = () => {
  return {
    type: types.FETCH_HOUSES_START,
  };
};

const fetchHousesSuccess = (housesList) => {
  return {
    type: types.FETCH_HOUSES_SUCCESS,
    housesList,
  };
};

const fetchHousesError = (e) => {
  return {
    type: types.FETCH_HOUSES_ERROR,
    error: e,
  };
};

//  choose house id

export const chooseHouse = (choosedHouseId) => {
  return {
    type: types.CHOOSE_HOUSE,
    choosedHouseId,
  };
};

// fetch house by id

export const fetchHouseById = (url) => {
  return async (dispatch) => {
    try {
      const response = await axios.get(url);
      dispatch(fetchHouseByIdSuccess(response.data));
    } catch (e) {
      console.log(e);
    }
  };
};

const fetchHouseByIdSuccess = (choosedHouse) => {
  return {
    type: types.FETCH_HOUSE_BY_ID,
    choosedHouse,
  };
};
