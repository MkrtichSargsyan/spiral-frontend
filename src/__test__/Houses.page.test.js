import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import HousesPage from '../pages/Houses.page';
import store from '../store/configureStore';

it('expect to render House page', () => {
  const result = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <HousesPage />
      </BrowserRouter>
    </Provider>,
  ).toJSON;
  expect(result).toMatchSnapshot();
});
