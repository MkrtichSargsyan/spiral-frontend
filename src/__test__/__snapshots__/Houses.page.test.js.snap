// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render House page 1`] = `[Function]`;
